src/main/java/com/hcl/appscan/jenkins/plugin/scanModes/PostmanCollection.java

```diff
@@ -0,0 +1,146 @@
+        m_additionalFiles = additionaFiles;
+    }
+
+    @DataBoundSetter
+    public void setPostmanCollectionFile(String postmanCollectionFile) {
+        m_postmanCollectionFile = postmanCollectionFile;
+    }
+
+    public String getPostmanCollectionFile() {
+        return m_postmanCollectionFile;
+    }
+
+    @DataBoundSetter
+    public void setAdditionalDomains(String additionalDomains) {
+        m_additionalDomains = additionalDomains;
+    }
+
+    public String getAdditionalDomains() {
+        return m_additionalDomains;
+    }
+
+    @DataBoundSetter
+    public void setEnvironmentalVariablesFile(String environmentalVariablesFile) {
+        m_environmentalVariablesFile = environmentalVariablesFile;
+    }
+
+    public String getEnvironmentalVariablesFile() {
+        return m_environmentalVariablesFile;
+    }
+
+    @DataBoundSetter
+    public void setGlobalVariablesFile(String globalVariablesFile) {
+        m_globalVariablesFile = globalVariablesFile;
+    }
+
+    public String getGlobalVariablesFile() {
+        return m_globalVariablesFile;
+    }
+
+    @DataBoundSetter
+    public void setAdditionalFiles(String additionalFiles) {
+        m_additionalFiles = additionalFiles;
+    }
+
+    public String getAdditionalFiles() {
+        return m_additionalFiles;
+    }
+
+
+    @Override
+    public Map<String, String> configureScanProperties(Map<String, String> properties, VariableResolver<String> resolver) {
+        properties.put("scanType", POSTMAN_COLLECTION);
+        if(isNullOrEmpty(m_postmanCollectionFile)) {
+            properties.put("postmanCollectionFile", resolver == null ? m_postmanCollectionFile : resolvePath(m_postmanCollectionFile, resolver));
+        }
+        if (isNullOrEmpty(m_additionalDomains)) {
+            properties.put("additionalDomains", resolver == null ? m_additionalDomains : Util.replaceMacro(m_additionalDomains, resolver));
+        }
+        if(isNullOrEmpty(m_environmentalVariablesFile)) {
+            properties.put("environmentalVariablesFile", resolver == null ? m_environmentalVariablesFile : resolvePath(m_environmentalVariablesFile, resolver));
+        }
+        if(isNullOrEmpty(m_globalVariablesFile)) {
+            properties.put("globalVariablesFile", resolver == null ? m_globalVariablesFile : resolvePath(m_globalVariablesFile, resolver));
+        }
+        if(isNullOrEmpty(m_additionalFiles)) {
+            properties.put("additionalFiles", resolver == null ? m_additionalFiles : resolvePath(m_additionalFiles, resolver));
+        }
+        return properties;
+    }
+
+    @Symbol("postman_collection") //$NON-NLS-1$
+    @Extension
+    public static final class DescriptorImpl extends ScanModeDescriptor {
+
+        @Override
+        public String getDisplayName() {
+            return "Postman Collection";
+        }
+
+        public FormValidation doCheckPostmanCollectionFile(@QueryParameter String postmanCollectionFile) {
```

Comment:
The 4 "doCheckXXXFile" all have very similar logic. I suggest moving that logic into a private method that all 4 of these methods call, rather than duplicating the logic 4 times.